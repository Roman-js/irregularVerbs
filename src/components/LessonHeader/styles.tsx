import {StyleSheet} from 'react-native';
import {colors} from '../../constants/colors';

export const styles = StyleSheet.create({
  wrapper: {
    display: 'flex',
    height: 100,
    backgroundColor: colors.lightBlue,
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
    paddingTop: 50,
    paddingHorizontal: 10,
  },
  backIcon: {
    transform: [{rotate: '90deg'}],
  },
  title: {
    color: colors.orange,
  },
});
