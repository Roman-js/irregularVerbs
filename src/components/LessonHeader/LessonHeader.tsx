import React from 'react';
import {useNavigation} from '@react-navigation/native';
import {StackNavigationProp} from '@react-navigation/stack';
import {Alert, TouchableOpacity, View} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import LeftChevronIcon from '../../assets/svg/LeftChevronIcon';
import SoundIcon from '../../assets/svg/SoundIcon';
import NoSoundIcon from '../../assets/svg/NoSoundIcon';
import {HomeTabParamList} from '../../navigations/HomeNavigation/HomeTab';
import {AppStateType} from '../../store/store';
import Typography from '../Typography';
import {styles} from './styles';
import {setSoundAvailableAction} from '../../store/actions/actions';
import {useTranslation} from 'react-i18next';

export const LessonHeader = () => {
  const navigation = useNavigation<StackNavigationProp<HomeTabParamList>>();
  const dispatch = useDispatch();
  const {t} = useTranslation();

  const {currentStep} = useSelector((states: AppStateType) => states.home);
  const {soundAvailable} = useSelector((states: AppStateType) => states.lesson);

  const goToThePrevPage = () => {
    Alert.alert(t('Main.areyousure2') as string, '', [
      {
        text: t('Main.stay') as string,
        style: 'cancel',
      },
      {
        text: t('Main.quit') as string,
        onPress: () => navigation.goBack(),
      },
    ]);
  };

  const onPressSoundSwitcher = () => {
    dispatch(setSoundAvailableAction());
  };

  const soundIcon = soundAvailable ? <SoundIcon /> : <NoSoundIcon />;

  return (
    <View style={styles.wrapper}>
      <TouchableOpacity onPress={goToThePrevPage}>
        <LeftChevronIcon style={styles.backIcon} />
      </TouchableOpacity>
      <Typography.H1 style={styles.title}>{currentStep}</Typography.H1>
      <TouchableOpacity onPress={onPressSoundSwitcher}>
        {soundIcon}
      </TouchableOpacity>
    </View>
  );
};
