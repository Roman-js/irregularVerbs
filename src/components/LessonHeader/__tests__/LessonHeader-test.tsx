import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import {LessonHeader} from '../LessonHeader';

it('Lesson Header snapShot', () => {
  const snap = renderer.create(<LessonHeader />).toJSON();
  expect(snap).toMatchSnapshot();
});
