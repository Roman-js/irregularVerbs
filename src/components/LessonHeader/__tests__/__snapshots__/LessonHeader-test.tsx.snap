// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lesson Header snapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D4E1FE",
      "display": "flex",
      "flexDirection": "row",
      "height": 100,
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingTop": 50,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="35"
      bbWidth="35"
      focusable={false}
      height="35"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "transform": Array [
              Object {
                "rotate": "90deg",
              },
            ],
          },
          Object {
            "flex": 0,
            "height": 35,
            "width": 35,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width="35"
    >
      <RNSVGGroup
        fill={
          Object {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGGroup
            fill={
              Object {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGGroup
              fill={
                Object {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGGroup
                fill={
                  Object {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGGroup
                  fill={
                    Object {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGPath
                    d="M0 0L24 0 24 24 0 24z"
                    fill={
                      Object {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M12 13.172L16.95 8.222 18.364 9.636 12 16 5.636 9.636 7.05 8.222z"
                    fill={
                      Object {
                        "payload": 4294942464,
                        "type": 0,
                      }
                    }
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 20,
          "fontWeight": "900",
          "lineHeight": 24,
        },
        Object {
          "color": "#D74214",
        },
      ]
    }
  >
    1
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      bbHeight="35"
      bbWidth="35"
      focusable={false}
      height={35}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 35,
            "width": 35,
          },
        ]
      }
      width={35}
    >
      <RNSVGGroup
        fill={
          Object {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M3 4.326c0 .729 1.134 2.39 2.519 3.692C8.587 10.9 8.27 12.337 4.418 13.01c-2.651.463-2.918.875-2.918 4.5 0 3.555.284 4.023 2.605 4.292 1.432.167 4.108 1.854 5.946 3.75C14.784 30.436 16 30.073 16 23.777v-5.223l6.775 6.723c6.56 6.51 9.225 8.087 9.225 5.459C32 29.105 5.87 3 4.237 3 3.556 3 3 3.597 3 4.326m23.531.779c-.35.887.34 3.257 1.541 5.292 1.704 2.888 2.071 4.76 1.698 8.644-.316 3.286-.051 5.311.784 6.004 1.877 1.557 3.808-5.144 3.012-10.451-.76-5.073-5.992-12.129-7.035-9.489M13 6.878c0 .482.675 1.437 1.5 2.122 1.253 1.04 1.5.896 1.5-.878C16 6.902 15.362 6 14.5 6c-.825 0-1.5.395-1.5.878m9.205 2.375c-.734.884-.623 1.823.378 3.191C23.362 13.51 24 15.331 24 16.491 24 18.605 25.7 21 27.2 21c1.561 0 .763-6.549-1.189-9.75-2.234-3.664-2.365-3.733-3.806-1.997"
          fill={
            Object {
              "payload": 4294942464,
              "type": 0,
            }
          }
          fillRule={0}
          propList={
            Array [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
